File: php/features.php
<?php
declare(strict_types=1);

// Pokapi SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class PokapiFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new PokapiBaseFeature();
            case "test":
                return new PokapiTestFeature();
            default:
                return new PokapiBaseFeature();
        }
    }
}

File: py/features.py
# Pokapi SDK feature factory

from feature.base_feature import PokapiBaseFeature
from feature.test_feature import PokapiTestFeature


def _make_feature(name):
    features = {
        "base": lambda: PokapiBaseFeature(),
        "test": lambda: PokapiTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
